GVFS/GVFS.Platform.Mac/MacPlatform.cs

```diff
@@ -170,6 +170,8 @@ public override string GVFSBinDirectoryName
 
             // Documented here (in the addressing section): https://www.unix.com/man-page/mojave/4/unix/
             public override int MaxPipePathLength => 104;
+
+            public override bool CaseSensitiveFileSystem => false;

```

Comment:
This is not exactly true. One can have Mac with a case-sensitive file system too. (see https://apple.stackexchange.com/questions/71357/how-to-check-if-my-hd-is-case-sensitive-or-not) 
Perhaps at clone time, VFS4G could establish this? Also, even on windows and linux, there are cases, where case-sensitivity might not match the default, see cases listed here: https://github.com/dotnet/corefx/issues/3156#issuecomment-171455461

Reply:
> One can have Mac with a case-sensitive file system too.

As best I could tell .NET Core has a mix of assumptions when it comes to Mac being case sensitive, which would make it difficult for us to support Mac file systems that are configured to be case sensitive.

https://github.com/dotnet/corefx/blob/e56e355d59380d7038040d57c01418f4950e47d0/src/Common/src/CoreLib/System/IO/Path.Unix.cs#L135
```
        /// <summary>Gets whether the system is case-sensitive.</summary>
        internal static bool IsCaseSensitive
        {
            get
            {
                #if PLATFORM_OSX
                    return false;
                #else
                    return true;
                #endif
            }
        }
```

https://github.com/dotnet/corefx/blob/e56e355d59380d7038040d57c01418f4950e47d0/src/Common/tests/Tests/System/IO/PathInternal.Unix.Tests.cs#L47
```
        [Fact]
        [PlatformSpecific(TestPlatforms.OSX)]
        public void IsCaseInsensitive_OSX()
        {
            // There have been reports of casing handling not being appropriate on MacOS
            // https://github.com/dotnet/corefx/issues/26797
            Assert.False(PathInternal.IsCaseSensitive);
        }
```

https://github.com/dotnet/corefx/blob/1bff7880bfa949e8c5e46039808ec412640bbb5e/src/Common/src/System/IO/PathInternal.CaseSensitivity.cs#L41

```
        /// <summary>
        /// Determines whether the file system is case sensitive.
        /// </summary>
        /// <remarks>
        /// Ideally we'd use something like pathconf with _PC_CASE_SENSITIVE, but that is non-portable, 
        /// not supported on Windows or Linux, etc. For now, this function creates a tmp file with capital letters 
        /// and then tests for its existence with lower-case letters.  This could return invalid results in corner 
        /// cases where, for example, different file systems are mounted with differing sensitivities.
        /// </remarks>
        private static bool GetIsCaseSensitive()
        {
            try
            {
                string pathWithUpperCase = Path.Combine(Path.GetTempPath(), "CASESENSITIVETEST" + Guid.NewGuid().ToString("N"));
                using (new FileStream(pathWithUpperCase, FileMode.CreateNew, FileAccess.ReadWrite, FileShare.None, 0x1000, FileOptions.DeleteOnClose))
                {
                    string lowerCased = pathWithUpperCase.ToLowerInvariant();
                    return !File.Exists(lowerCased);
                }
            }
            catch (Exception exc)
            {
                // In case something goes terribly wrong, we don't want to fail just because
                // of a casing test, so we assume case-insensitive-but-preserving.
                Debug.Fail("Casing test failed: " + exc);
                return false;
            }
        }
```
To prevent someone from getting in a bad state #1384 updates the clone/mount verbs to fail if someone tries to use them on a Mac case sensitive filesystem.

Reply:
I was going to make the same observation that #1384 prevents mounting on case-sensitive Mac filesystems -- at least for now.

Reply:
Ok, if there's inconsistent support for case sensitive Mac file system in dotnet core itself, then it makes sense to not try to support them here.